src/main/telemetry/ghst.c

```diff
@@ -153,6 +199,19 @@ void initGhstTelemetry(void)
         || (isAmperageConfigured() && telemetryIsSensorEnabled(SENSOR_CURRENT | SENSOR_FUEL))) {
         ghstSchedule[index++] = BIT(GHST_FRAME_PACK_INDEX);
     }
+
+ #ifdef USE_GPS
+    if (featureIsEnabled(FEATURE_GPS)
```

Comment:
Just checking if the user has enabled the GPS feature doesn't mean the GPS is working. So might want to consider `sensors(SENSOR_GPS)` which means the sensor is actually working otherwise the state of the `gpsSol` global is not really defined. But it's slightly more complicated then checking once during initialization since it can take some time for the sensor state to become active and/or it can change state later (like if the GPS module stops responding).

Also should probably also consider `STATE(GPS_FIX)` at some point. Without a fix the `gpsSol` position data is not useful.

Reply:
Note that continuing to send telemetry frames after `STATE(GPS_FIX)` is false negates some fo the value of OpenTx telemetry. Take the example that the user crashes upside down in a field. The GPS module begins to drop satellites and eventually loses the 3D fix. As the fix is lost the GPS module still continues to supply position information but it rapidly becomes extremely inaccurate - quickly reaching miles away.

So if the telemetry packet continues to be sent then the user's radio will show this inaccurate position information making recovery difficult. However behaving like Smartport telemetry where the GPS telemetry frame is not sent when there is no GPS fix causes the sensor to be lost in OpenTx which results in the last position to remain displayed on the user's radio.

Reply:
Another good point. The STATE(GPS_FIX) is a good suggestion, that I can add in one of the unused fields in the secondary GPS packet. 
For the first part of your comment though, is there a callback or other notification available when a sensor arrives or departs? would be preferable to polling each time telemetry is sent.

Reply:
No, there is no "callback" on the sensor state. In other cases (like the OSD) we keep a local static of the state and compare each iteration looking for a change event to `sensors(SENSOR_GPS)`.

Reply:
> Note that continuing to send telemetry frames after `STATE(GPS_FIX)` is false negates some fo the value of OpenTx telemetry. Take the example that the user crashes upside down in a field. The GPS module begins to drop satellites and eventually loses the 3D fix. As the fix is lost the GPS module still continues to supply position information but it rapidly becomes extremely inaccurate - quickly reaching miles away.
> 
> So if the telemetry packet continues to be sent then the user's radio will show this inaccurate position information making recovery difficult. However behaving like Smartport telemetry where the GPS telemetry frame is not sent when there is no GPS fix causes the sensor to be lost in OpenTx which results in the last position to remain displayed on the user's radio.

The problem with this is that before the 3d fix is finally lost, the PDOP degrades, and already looses these critical last low-PDOP (high accuracy) points. 

A better way to deal with this (at least in the case when finding a downed model) is to keep around the last few points on the OpenTx side, along with their PDOP (or just satellite count), and stop logging them when the 3d fix drops out. 

This then brings up the flip side of not sending telemetry before a 3d fix is obtained. If you don't have an OSD on which to view the satellite count, and you don't pass this data to the radio before 3d fix, then you have no way to debug low sat-count startups. The satellite count data is just not available anywhere. 

Which leads me to wonder whether the correct solution is to send the 'raw' GPS data, including satellite count, along with the 3D fix status, and let the radio/user determine how to interpret that data. If the user just wants to watch the satellite count increase healthily until a 3D fix is acquired, then why not. 
If the user wants to setup vocal alerts on the radio counting as each satellite is acquired, then also, why not.

Reply:
The best solution is what I have implemented for Smartport telemetry. I haven't submitted the improvements yet since they depend on another unmerged PR (#10566) and the addition of `STATE(GPS_FIX_EVER)`. But you can see them here: https://github.com/etracer65/betaflight/commit/69e69e4447d2b53673f4c6b7ea494d1142936b62

Basically the logic sends "dummy" GPS frames (with zeros) until the first fix is obtained, then after that only when there's an active fix. This makes it easy for the user to discover sensors (since they don't have to wait for a fix) and get to see the satellite count "grow". After they have a fix and subsequently lose the fix then we stop sending GPS frames so they keep the last fix on their radio.

Reply:
For the flags, looks like the two interesting ones are GPS_FIX, and GPS_FIX_HOME, which rolling in the distance/direction to home data results in this code for the secondary telemetry packet. 
Distance is transferred in units of 10m to fit in a U16 and give a range of 655km, since 65km is quite FPV'able these days. 

`// GPS data, secondary, auxiliary data
void ghstFrameGpsSecondaryTelemetry(sbuf_t *dst)
{
    // use sbufWrite since CRC does not include frame length
    sbufWriteU8(dst, GHST_FRAME_GPS_PAYLOAD_SIZE + GHST_FRAME_LENGTH_CRC + GHST_FRAME_LENGTH_TYPE);
    sbufWriteU8(dst, GHST_DL_GPS_SECONDARY);

    sbufWriteU16(dst, gpsSol.groundSpeed);      // speed in 0.1m/s
    sbufWriteU16(dst, gpsSol.groundCourse);     // degrees * 10
    sbufWriteU8(dst, gpsSol.numSat);
	
    sbufWriteU16(dst, (uint16_t) (GPS_distanceToHome / 10));    // use units of 10m to increase range of U16 to 655.36km
    sbufWriteU16(dst, GPS_directionToHome);

    uint8_t gpsFlags = 0;
    if(STATE(GPS_FIX))
        gpsFlags |= GPS_FLAGS_FIX;
    if(STATE(GPS_FIX_HOME))
        gpsFlags |= GPS_FLAGS_FIX_HOME;
    sbufWriteU8(dst, gpsFlags);
}`